source/develop/release-management/features/network/sriov-live-migration.md

```diff
@@ -0,0 +1,172 @@
+
+The current implementation allowed to mitigate this downtime
+[Live Migration Support For SRIOV][1], however this approach required manual
+configuration to be done on a VM which is not convenient for large setups. 
+
+In order to achieve minimal downtime the [NET_FAILOVER][4] support was added 
+to Linux kernel. The driver provides automated mechanism which is transparent 
+to the VM, and the user space. It is achieved by `failover` netdev which acts
+as master device and controls two slave devices, the `primary` device being 
+SR-IOV passtrough and the virtio as `standby`. The VM user access network
+through the `failover` device. 
+
+There is still need to provide two interfaces, the SR-IOV passtrough device
+and virtio device, to the affected VM, but the  overall configuration overhead
+has been reduced to selecting suitable vnic profile pair.
+
+
+## Prerequisites
+
+The implementation requires a [Linux kernel][5], [libvirt][6] and [qemu][7] 
+support with `net_failover`. 
+
+The [Linux kernel][5] support has to be only on guest side. 
+
+## Limitations
+
+* The virtio network has to be required to ensure connectivity on the source,
+  and the destination hosts.
+
+## Implementation
+
+### Entity Description
+
+Vm nic will get a new attribute that will point to the failover virtio 
+vnic profile. 
+
+### User Experience
+
+Upon creation of new VM nic, the user can enable failover and select 
+the failover network vnic profile. The `failover` selection is available 
+only for passthrough master vnic profiles. 
+
+REST API takes an additional parameter for VM nic called `failover`
+containing id of the failover vnic profile:
+
+```
+<nic>
+  <name>mynic</name>
+  <vnic_profile id='789'/>
+  <failover id='123'/>
+</nic>
+```
+
+### Engine backend
+
+As engine is responsible for creating XML for a new VM nic, the 
+appropriate XML will be created and provided to Libvirt. 
+
+In order to ensure that the virtio network is available it has to be
+marked as required, and the migration policy needs to be updated.
+
+### Libvirt XML
+
+If the feature is activated for a new VM nic, the libvirt XML definition
+contains an additional network with attribute [teaming element][8],
+as follows:
+
+```
+<devices>
+  <interface type='network'>
+    <source network='mybridge'/>
+    <mac address='00:11:22:33:44:55'/>
+    <model type='virtio'/>
+    <teaming type='persistent'/>
+    <alias name='ua-backup0'/>
+  </interface>
+  <interface type='network'>
+    <source network='hostdev-pool'/>
+    <mac address='00:11:22:33:44:55'/>
+    <model type='virtio'/>
+    <teaming type='transient' persistent='ua-backup0'/>
+  </interface>
+</devices>
+```
+
+### Vdsm
+
+Handling of update and device removal is handled in vdsm. In both cases we 
+can actually apply changes that are relevant to both interface as those 
+interfaces are linked by the `teaming` attribute. 
+
+Relevant update changes are:
+* Nic plug/unplug
+* Nic being linked
```

Comment:
I think link/unlink is not supported for SR-IOV.

Reply:
Right